src/netspresso_trainer/cfg/augmentation.py

```diff
@@ -15,56 +23,45 @@ class ColorJitter:
 
 
 @dataclass
-class AugmentationConfig:
-    img_size: int = 256
-    max_scale: Optional[int] = 1024
-    min_scale: Optional[int] = None
-    crop_size_h: Optional[int] = None
-    crop_size_w: Optional[int] = None
-    resize_ratio0: Optional[float] = None
-    resize_ratiof: Optional[float] = None
-    resize_add: Optional[float] = 1
-    fliplr: Optional[float] = 0.5
-    color_jitter: Optional[ColorJitter] = field(default_factory=lambda: ColorJitter())
-    
-    
+class Pad(Transform):
+    name: str = 'pad'
+    padding: Union[int, List] = 0
+
 
 @dataclass
-class ClassificationAugmentationConfig(AugmentationConfig):
-    resize_ratio0 = None
-    resize_ratiof = None
-    resize_add = None
-    color_jitter = None
+class RandomCrop(Transform):
+    name: str = 'randomcrop'
+    size: int = DEFAULT_IMG_SIZE
+    interpolation: Optional[str] = 'bilinear'
 
 
 @dataclass
-class SegmentationAugmentationConfig(AugmentationConfig):
-    img_size = 512
-    resize_ratio0 = 1.0
-    resize_ratiof = 1.5
-    
-    def __post_init__(self):
-        # variable interpolation
-        if self.min_scale is None:
-            self.min_scale = self.img_size
-        if self.crop_size_h is None:
-            self.crop_size_h = self.img_size
-        if self.crop_size_w is None:
-            self.crop_size_w = self.img_size
-    
+class RandomResizedCrop(Transform):
+    name: str = 'randomresizedcrop'
+    size: int = DEFAULT_IMG_SIZE
+    interpolation: Optional[str] = 'bilinear'
+
 
 @dataclass
-class DetectionAugmentationConfig(AugmentationConfig):
-    img_size = 512
-    max_scale = 2048
-    min_scale = 768
-    resize_ratio0: 0.5
-    resize_ratiof: 2.0
-    resize_add: 1
-    
-    def __post_init__(self):
-        # variable interpolation
-        if self.crop_size_h is None:
-            self.crop_size_h = self.img_size
-        if self.crop_size_w is None:
-            self.crop_size_w = self.img_size
\ No newline at end of file
+class RandomHorizontalFlip(Transform):
+    name: str = 'randomhorizontalflip'
+    p: float = 0.5
+
+
+@dataclass
+class RandomVerticalFlip(Transform):
+    name: str = 'randomverticalflip'
+    p: float = 0.5
+
+
+@dataclass
+class Resize(Transform):
+    name: str = 'resize'
+    size: int = DEFAULT_IMG_SIZE
+    interpolation: Optional[str] = 'bilinear'
+
+
+@dataclass
+class AugmentationConfig:
+    img_size: int = DEFAULT_IMG_SIZE
+    recipe: List[Transform] = field(default_factory=list)
```

Comment:
@illian01 

Cuz there is a type hint with `List[Transform]`, it seems better to set `default_factory` as returning basic Transform instance wrapped with the list.

We cannot guarantee the type of `recipe`.

Reply:
Good. I changed default to `[Transform()]`